src/supervised/distance.jl

```diff
@@ -108,32 +108,88 @@ islipschitzcont_deriv(::L2DistLoss) = true
 isconvex(::L2DistLoss) = true
 isstronglyconvex(::L2DistLoss) = true
 
+
+# ===========================================================
+# L(y, t) = 1 - cos((y-t)*2π/c)
+
+"""
+`PeriodicLoss <: DistanceLoss`
+
+Measures distance on a circle of specified circumference.
+""" 
+immutable PeriodicLoss{T<:AbstractFloat} <: DistanceLoss
+    k::T   # k = 2π/circumference
+    function PeriodicLoss(circ::T)
+        circ > 0 || error("circumference should be strictly positive")
+        new(convert(T, 2π/circ))
+    end
+end
+function PeriodicLoss{T<:Number}(ε::T=1.0)
+    if T <: AbstractFloat
+        PeriodicLoss{T}(ε)
+    else # cast to Float64
+        PeriodicLoss{Float64}(Float64(ε))
+    end
+end
```

Comment:
What was wrong with the two method approach? Did the compiler complain about ambiguity ?

```julia
PeriodicLoss{T<:AbstractFloat}(circ::T=1.0) = PeriodicLoss{T}(circ)
PeriodicLoss(circ) = PeriodicLoss{Float64}(Float64(circ))
``` 

looks a bit nicer